common/ast/conversion.go

```diff
@@ -649,6 +649,21 @@ func ProtoConstantAsVal(c *celpb.Constant) (ref.Val, error) {
 	return nil, fmt.Errorf("unsupported constant kind: %v", c.GetConstantKind())
 }
 
+// ProtoStructValueAsVal converts a canonical structpb.Value protobuf to a CEL-native ref.Val.
+func ProtoStructValueAsVal(c *structpb.Value) (ref.Val, error) {
```

Comment:
You should just be able to use `types.DefaultTypeAdapter.NativeToValue()` and have this work correctly.